Core/TrackerDataManager.swift

```diff
@@ -0,0 +1,95 @@
+//
+//  TrackerData.swift
+//  Core
+//
+//  Copyright © 2019 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+
+public class TrackerDataManager {
+    
+    public static let shared = TrackerDataManager()
+    
+    private(set) public var trackerData: TrackerData!
+    
+    init() {
+        reload()
+    }
+    
+    public func reload() {
+        // valid use case is we don't have downloaded data yet
+        let data = FileStore().loadAsData(forConfiguration: .trackerDataSet) ?? Self.loadEmbeddedAsData()
+        do {
+            self.trackerData = try JSONDecoder().decode(TrackerData.self, from: data)
+        } catch {
+            Logger.log(text: "error loading downloaded tds [\(error.localizedDescription)], falling back to embedded")
+            
+            // The embedded data should NEVER fail, so fall back to it
+            let embeddedData = try? JSONDecoder().decode(TrackerData.self, from: Self.loadEmbeddedAsData())
+            self.trackerData = embeddedData!
+            Pixel.fire(pixel: .reloadTrackerDataFailed)
```

Comment:
You can use Pixel's `fire(pixel: PixelName, error: Error) ` method to embed error info as parameters.